plugins/inputs/gcs/gcs.go

```diff
@@ -0,0 +1,307 @@
+package gcs
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"os"
+
+	"cloud.google.com/go/storage"
+	"golang.org/x/oauth2/google"
+	"google.golang.org/api/iterator"
+	"google.golang.org/api/option"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/plugins/inputs"
+	"github.com/influxdata/telegraf/plugins/parsers"
+)
+
+const (
+	emulatorHostEnv  = "STORAGE_EMULATOR_HOST"
+	defaultOffSetKey = "offset-key.json"
+)
+
+type GCS struct {
+	CredentialsFile string `toml:"credentials_file"`
+	Project         string `toml:"project"`
+	Bucket          string `toml:"bucket"`
+
+	Prefix              string `toml:"key_prefix"`
+	OffsetKey           string `toml:"offset_key"`
+	ProcessOffset       bool   `toml:"process_offset"`
```

Comment:
```suggestion
```
This boolean doesn't seem to be used, should it be removed?

Reply:
Done